plugins/module_utils/network/sonic/config/l3_interfaces/l3_interfaces.py

```diff
@@ -256,521 +296,491 @@ def _state_deleted(self, want, have):
+        ip, mask = ip_prefix.split('/')
+        addr_payload = {
+            'ip': ip,
+            'openconfig-if-ip:config': {
+                'ip': ip,
+                'prefix-length': int(mask)
+            }
+        }
         if secondary:
-            cfg['secondary'] = secondary
-        if ipv6_eui64:
-            cfg['openconfig-interfaces-private:eui64'] = ipv6_eui64
-        addr_payload = {'ip': ip, 'openconfig-if-ip:config': cfg}
+            addr_payload['openconfig-if-ip:config']['secondary'] = secondary
         return addr_payload
 
-    def get_sub_interface_name(self, name):
-        sub_intf = "subinterfaces/subinterface=0"
-        if name.startswith("Vlan"):
-            sub_intf = "openconfig-vlan:routed-vlan"
-        return sub_intf
+    @staticmethod
+    def get_ipv6_addr_payload(ipv6_prefix, eui64=False):
+        ipv6, mask = ipv6_prefix.split('/')
+        addr_payload = {
+            'ip': ipv6,
+            'openconfig-if-ip:config': {
+                'ip': ipv6,
+                'prefix-length': int(mask)
+            }
+        }
+        if eui64:
+            addr_payload['openconfig-if-ip:config']['openconfig-interfaces-private:eui64'] = eui64
+        return addr_payload
 
-    def build_update_ipv6_enabled(self, ipv6_enabled):
-        payload = {'config': {'enabled': ipv6_enabled}}
-        return payload
+    @staticmethod
+    def remove_defaults(have_conf, conf=None, delete_op=True):
+        if delete_op:
+            # For delete operation, the default values in have_conf are removed
+            updated_conf = have_conf.copy()
+            if have_conf and have_conf.get('ipv6'):
+                updated_conf['ipv6'] = have_conf['ipv6'].copy()
+                for option in ('enabled', 'autoconf', 'dad'):
+                    if updated_conf['ipv6'].get(option) == DEFAULT_VALUES['ipv6'][option]:
+                        del updated_conf['ipv6'][option]
+
+                if not updated_conf['ipv6']:
+                    del updated_conf['ipv6']
+        else:
+            # For merge operation, the default values in conf are removed
+            # if that option is not present in have_conf
+            updated_conf = conf.copy()
+            if conf and conf.get('ipv6'):
+                have_ipv6 = have_conf.get('ipv6', {}) if have_conf else {}
+                updated_conf['ipv6'] = conf['ipv6'].copy()
+                for option in ('enabled', 'autoconf', 'dad'):
+                    if option not in have_ipv6 and conf['ipv6'].get(option) == DEFAULT_VALUES['ipv6'][option]:
+                        del updated_conf['ipv6'][option]
+
+                if not conf['ipv6']:
+                    del conf['ipv6']
+
+        return updated_conf
+
+    def get_diff(self, want, have):
+        updated_want = []
+        for conf in want:
+            if conf.get('ipv6'):
+                have_conf = next((item for item in have if item['name'] == conf['name']), {})
```

Comment:
dict comprehension could also be used here to reduce the time spent determining "have_conf".